wgengine/wglog/wglog.go

```diff
@@ -43,25 +42,26 @@ func NewLogger(logf logger.Logf) *Logger {
 			// See https://github.com/tailscale/tailscale/issues/1388.
 			return
 		}
-		msg := fmt.Sprintf(format, args...)
-		r := ret.replacer.Load()
-		if r == nil {
+		replace, _ := ret.replace.Load().(map[string]string)
+		if replace == nil {
 			// No replacements specified; log as originally planned.
 			logf(format, args...)
 			return
 		}
 		// Do the replacements.
-		new := r.(*strings.Replacer).Replace(msg)
-		if new == msg {
-			// No replacements. Log as originally planned.
-			logf(format, args...)
-			return
+		for i, arg := range args {
+			peer, ok := arg.(*device.Peer)
+			if !ok {
+				continue
+			}
+			wgStr := peer.String()
+			tsStr, ok := replace[wgStr]
+			if !ok {
+				continue
+			}
+			args[i] = tsStr
```

Comment:
technically this isn't our memory to mutate. It's probably usually safe because the slice backing array is allocated by the compiler, though.

maybe restore these at the end at least? little less wrong.

Reply:
Hrm. Yeah, I'll make a copy of the slice. I may be able to make that cheap...